MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -32,340 +30,299 @@ namespace MediaBrowser.MediaEncoding.Encoder
-        }
-
-        private void ConfigureEncoderPaths()
-        {
-            if (_hasExternalEncoder)
+            try
             {
-                return;
-            }
-
-            var appPath = GetEncodingOptions().EncoderAppPath;
+                var files = FileSystem.GetFilePaths(path);
 
-            if (string.IsNullOrWhiteSpace(appPath))
-            {
-                appPath = Path.Combine(ConfigurationManager.ApplicationPaths.ProgramDataPath, "ffmpeg");
-            }
+                var excludeExtensions = new[] { ".c" };
 
-            var newPaths = GetEncoderPaths(appPath);
-            if (string.IsNullOrWhiteSpace(newPaths.Item1) || string.IsNullOrWhiteSpace(newPaths.Item2) || IsSystemInstalledPath(appPath))
-            {
-                newPaths = TestForInstalledVersions();
+                return files.FirstOrDefault(i => string.Equals(Path.GetFileNameWithoutExtension(i), filename, StringComparison.OrdinalIgnoreCase)
+                                                    && !excludeExtensions.Contains(Path.GetExtension(i) ?? string.Empty));
             }
-
-            if (!string.IsNullOrWhiteSpace(newPaths.Item1) && !string.IsNullOrWhiteSpace(newPaths.Item2))
+            catch (Exception)
             {
-                FFMpegPath = newPaths.Item1;
-                FFProbePath = newPaths.Item2;
+                // Trap all exceptions, like DirNotExists, and return null
+                return null;
             }
-
-            LogPaths();
         }
 
-        private Tuple<string, string> GetEncoderPaths(string configuredPath)
+        private string GetProbePathFromEncoderPath(string appPath)
         {
-            var appPath = configuredPath;
-
-            if (!string.IsNullOrWhiteSpace(appPath))
+            if (!string.IsNullOrEmpty(appPath))
             {
-                if (Directory.Exists(appPath))
-                {
-                    return GetPathsFromDirectory(appPath);
-                }
-
-                if (File.Exists(appPath))
-                {
-                    return new Tuple<string, string>(appPath, GetProbePathFromEncoderPath(appPath));
-                }
-            }
-
-            return new Tuple<string, string>(null, null);
-        }
+                string pattern = @"[^\/\\]+?(\.[^\/\\\n.]+)?$";
+                string substitution = @"ffprobe$1";
 
-        private Tuple<string, string> TestForInstalledVersions()
-        {
-            string encoderPath = null;
-            string probePath = null;
+                var options = RegexOptions.Multiline;
+                var regex = new Regex(pattern, options);
 
-            if (_hasExternalEncoder && ValidateVersion(_originalFFMpegPath, true))
-            {
-                encoderPath = _originalFFMpegPath;
-                probePath = _originalFFProbePath;
+                return regex.Replace(appPath, substitution);
```

Comment:
This is called multiple time, right?
Use the static Regex instead of a new instance.

Reply:
no it's called once per discovery.  Shall change to return Regex.Replace()